src/com/google/cose/utils/CoseUtils.java

```diff
@@ -45,4 +66,93 @@ public static DataItem encodeStructure(String context, Map protectedBodyHeaders,
     }
     return arrayBuilder.end().build().get(0);
   }
+
+  public static LinkedHashMap<Integer, DataItem> getLabelsFromMap(Map keyMap)
+      throws CborException {
+    final LinkedHashMap<Integer, DataItem> labels = new LinkedHashMap<>();
+    for (DataItem item : keyMap.getKeys()) {
+      labels.put(CborUtils.asInteger(item), keyMap.get(item));
+    }
+    return labels;
+  }
+
+  /**
+   * Returns DataItem from a cbor map based on Integer index.
+   * @param cborMap map that has the information.
+   * @param index integer index to be used as key in the map.
+   * @return value in the map corresponding to key
+   */
+  public static DataItem getValueFromMap(Map cborMap, int index) {
+    final Number key;
+    if (index >= 0) {
+      key = new UnsignedInteger(index);
+    } else {
+      key = new NegativeInteger(index);
+    }
+    return cborMap.get(key);
+  }
+
+  /**
+   * Generates ECC Private Key from d coordinate value.
+   *
+   * Only supports P256 curve currently.
+   * @param curve supported curve
+   * @param d raw bytes for the private key coordinate.
+   * @return PrivateKey JCA implementation
+   * @throws CoseException if unsupported key curve is used.
+   */
+  public static PrivateKey generateEccPrivateKey(int curve, BigInteger d) throws CoseException {
+    try {
+      if (d == null) {
```

Comment:
Why not an exception on null? (like `generateEccPublicKey`)

Reply:
yeah.. good catch. Not sure why this happened 😅